internal/store/app.go

```diff
@@ -0,0 +1,70 @@
+package store
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/deislabs/duffle/pkg/claim"
+	"github.com/deislabs/duffle/pkg/utils/crud"
+	"github.com/pkg/errors"
+)
+
+const (
+	// AppConfigDirectory is the Docker App directory name inside Docker config directory
+	AppConfigDirectory = "app"
+	// BundleStoreDirectory is the bundle store directory name
+	BundleStoreDirectory = "bundles"
+	// CredentialStoreDirectory is the credential store directory name
+	CredentialStoreDirectory = "credentials"
+	// ClaimStoreDirectory is the claim store directory name
+	ClaimStoreDirectory = "claims"
+)
+
+// ApplicationStore is the main point to access different stores:
+// - Bundle store persists all bundles built or fetched locally
+// - Credential store persists all the credentials, per context basis
+// - Claim store persists all the claims, per context basis
+type ApplicationStore struct {
+	path string
+}
+
+// NewApplicationStore creates a new application store, nested inside a
+// docker configuration directory. It will create all the directory hierarchy
+// if anything is missing.
+func NewApplicationStore(configDir string) (*ApplicationStore, error) {
+	storePath := filepath.Join(configDir, AppConfigDirectory)
+	directories := []string{BundleStoreDirectory, CredentialStoreDirectory, ClaimStoreDirectory}
+	for _, d := range directories {
+		if err := os.MkdirAll(filepath.Join(storePath, d), 0755); err != nil {
+			return nil, errors.Wrapf(err, "failed to create application store directory %q", d)
+		}
+	}
+	return &ApplicationStore{path: storePath}, nil
+}
+
+// ClaimStore initializes and returns a context based claim store
+func (a ApplicationStore) ClaimStore(context string) (ClaimStore, error) {
+	path := filepath.Join(a.path, ClaimStoreDirectory, context)
```

Comment:
what if the context has special characters, or even is named ".."? to avoid this the cli context store uses `sha256(contextName)` as the directory name